### PR TITLE
Native stack walking with DWARF v2

### PR DESCRIPTION
**Pull request by javierhonduco**
This PR implements the v2 of the DWARF stack unwinder:

- The first commit reduces the unwind table size by half, allowing for more entries to fit in the table;
- The second commit removes `bpf_loop`, reducing the minimum required kernel version;
- The third commit reenables the CPU tests;

The details for each commit can be found in their commit message.

## Test Plan

Ensured, visually, that the stacks are correct:

**`testdata/out/basic-cpp-no-fp`**:

<details>
image
</details>

**CRuby interpreter**

<details>
image
[...]
image
</details>

### CODE REVIEW

File: pkg/profiler/cpu/maps.go [hunk 1]
```diff
@@ -167,37 +181,46 @@ func (m *bpfMaps) setUnwindTable(pid int, ut unwind.UnwindTable) error {
...
+			return fmt.Errorf("write CFA register bytes: %w", err)
+		}
+
 		// Write CFA.
 		switch row.CFA.Rule {
 		case frame.RuleCFA:
 			// Write CFA register.
-			if err := binary.Write(buf, m.byteOrder, row.CFA.Reg); err != nil {
+			var CFARegister uint16
+			if row.CFA.Reg == 6 {
```
Comment (kakkoyun) [inline pkg/profiler/cpu/maps.go:194]: nit: Let's use some constants instead of magic numbers here.

File: bpf/cpu/cpu.bpf.c [hunk 2]
```diff
@@ -252,117 +268,102 @@ bpf_map_lookup_or_try_init(void *map, const void *key, const void *init) {
...
-  u64 table_len = unwind_table->table_len;
-  // Just for debugging to ensure that the data we are reading
-  // matches what we wrote.
-  bpf_printk("- unwind table has %d items", table_len);
 
-  // Invariant check.
-  if (table_len >= MAX_UNWIND_TABLE_SIZE) {
-    bpf_printk("should never happen");
-    BUMP_UNWIND_SHOULD_NEVER_HAPPEN_ERROR();
+SEC("perf_event")
```
Comment (kakkoyun) [inline bpf/cpu/cpu.bpf.c:326]: Why is this needed? I always thought it was for hooking to the event. Does it also use for exporting functions?

**Review by kakkoyun — COMMENTED** [review]
Awesome job! 
v2 is here even before we released the v1!

I've some minor comments.

File: pkg/profiler/cpu/maps.go (hunk 1 above)
Comment (javierhonduco) [reply pkg/profiler/cpu/maps.go:194]: Good shout Was planning to do it but it seems I forgot about it

File: bpf/cpu/cpu.bpf.c (hunk 2 above)
Comment (javierhonduco) [reply bpf/cpu/cpu.bpf.c:326]: Yes, all the other functions are inlined and one single program is executed. The compiler has replaced every call site with its body and they don't really exist as functions or individual programs anymore. 

As we are doing a tail call to a function, this has to be present and has to be the same BPF type as the BPF program that invokes it

**Review by kakkoyun — APPROVED** [review]
LGTM

**Comment by zdyj3170101136** [comment]
is cfa_offset int16 enough?

it represent the stack size, int32 is recommend?

**Comment by brancz** [comment]
Note that this code is entirely replaced starting v0.32
